tools/validators/instance_validator/instance_validator.py

```diff
@@ -43,17 +43,26 @@ def _ParseArgs() -> argparse.ArgumentParser:
     ArgumentParser object
   """
   parser = argparse.ArgumentParser(
-      description='Validate a YAML building configuration file')
+      description='Validate YAML building configuration files')
 
   parser.add_argument(
       '-i',
       '--input',
       action='append',
       dest='filenames',
-      required=True,
-      help='Filepaths to YAML building configurations',
+      required=False,
+      help='Local path to a YAML building configuration file. May be used multiple times.',
       metavar='FILE')
 
+  parser.add_argument(
+    '-d',
+    '--dir-input',
+    action='append',
+    dest='directories',
+    required=False,
+    help='Local path to a directory (folder) containing YAML building configuration files. May be used multiple times.',
+    metavar='DIR')
```

Comment:
these could be made part of a mutually exclusive group https://stackoverflow.com/questions/11154946/require-either-of-two-arguments-using-argparse

Or even better, let the user specify a path that could be a file or a directory - it's easy to tell which it is and handle it in a loop.  That way you don't have to add an argument to all the method signatures down to deserialization

Reply:
This is a good point and idea!